### PR TITLE
Redirect bugfix

### PR DESCRIPTION
**Pull request by GregHanson**
modified PR of https://github.com/istio/istio/pull/2610, cherrypicked the test commit and cleaned up the bug fix logic

### CODE REVIEW

File: pilot/proxy/envoy/config.go [hunk 1]
```diff
@@ -895,8 +895,10 @@ func buildEgressVirtualHost(rule *routing.EgressRule,
 		// Set the destination clusters to the cluster we computed above.
 		// Services defined via egress rules do not have labels and hence no weighted clusters
 		for _, route := range routes {
-			route.Cluster = externalTrafficCluster.Name
-			route.clusters = []*Cluster{externalTrafficCluster}
+			if !route.Redirect() {
+				route.Cluster = externalTrafficCluster.Name
+				route.clusters = []*Cluster{externalTrafficCluster}
+			}
```
Comment (rshriram) [inline pilot/proxy/envoy/config.go]: nice!

File: pilot/test/integration/infra.go [hunk 2]
```diff
@@ -483,10 +483,6 @@ func (infra *infra) deleteConfig(inFile string) error {
 			return err
 		}
 	}
-
-	sleepTime := time.Second * 3
```
[the remark this answers is not in the document]
Comment (rshriram) [reply pilot/test/integration/infra.go]: Yeah, don't remove this. Don't mind the extra delay, in exchange for reliability.

[the remark this answers is not in the document]
Comment (ronenschafferibm) [reply pilot/test/integration/infra.go]: I want to mention that the test is also sleeping for 3 sec after applying each config.
Moreover, at Teardown the test calls deleteAllConfigs() and doesn't sleep at all.

**Pull request merged**